src/api/app/models/workflow_run.rb

```diff
@@ -27,10 +28,25 @@ class WorkflowRun < ApplicationRecord
     fail: 2
   }
 
-  def update_to_fail(message)
+  # Marks the workflow run as failed and records the relevant debug information in response_body
+  def update_as_failed(message)
     update(response_body: message, status: 'fail')
   end
 
+  # Stores debug info to help figure out what went wrong when trying to save a Status in the SCM.
+  def report_failure_to_scm(message, options)
```

Comment:
@danidoni I find the naming a bit confusing, I was expecting that this method is reporting something to the scm, not saving the report in OBS.

Reply:
maybe something like `save_failed_scm_status_report`

Reply:
Yes, perfect! Changed